qcodes/instrument/base.py

```diff
@@ -93,11 +93,26 @@ def add_parameter(self, name: str,
             **kwargs: Constructor arguments for ``parameter_class``.
 
         Raises:
-            KeyError: If this instrument already has a parameter with this
-                name.
-        """
-        if name in self.parameters:
-            raise KeyError(f'Duplicate parameter name {name}')
+            ValueError: If this instrument already has a parameter with this
```

Comment:
Chancing this from a KeyError to a ValueError is technically an API break but I am not sure we care? @astafan8 What do you think ?